mne_bids/tests/test_mne_bids.py

```diff
@@ -111,9 +118,16 @@ def test_ctf():
 
     raw_to_bids(subject_id=subject_id, session_id=session_id, run=run,
                 task=task, acquisition=acq, raw_file=raw_fname,
-                output_path=output_path, overwrite=True)
+                output_path=output_path, write_mode='append')
     cmd = ['bids-validator', output_path]
     run_subprocess(cmd, shell=shell)
+
+    # test to check that the 'error' write_mode parameter works correctly
+    with pytest.raises(OSError, match="already exists"):
+        raw_to_bids(subject_id=subject_id, session_id=session_id, run=run,
+                    task=task, acquisition=acq, raw_file=raw_fname,
+                    output_path=output_path, write_mode='error')
```

Comment:
can you also add this test for another file format -- like EEGLAB or vhdr? The ones that use the `copyfile_xxx` functions

Reply:
yep, can do. I am just re-writing some of the logic for the error code for the files which can have data appended which I think will hopefully make it a bit better...